src/plugins/share/README.mdx

```diff
@@ -0,0 +1,243 @@
+---
+id: kibDevSharePluginReadme
+slug: /kibana-dev-docs/tutorials/share
+title: Kibana share Plugin
+summary: Introduction to Kibana "share" plugin services (sharing popup menu, URL locators, and short URLs).
+date: 2022-02-21
+tags: ['kibana', 'onboarding', 'dev', 'tutorials', 'share', 'url', 'short-url', 'locator']
+---
+
+# Share plugin
+
+The `share` plugin contains various utilities for displaying sharing context menu,
+generating deep links to other apps using *locators*, and creating short URLs.
+
+
+## Sharing context menu
+
+You can register an item into sharing context menu (which is displayed in
+Dashboard, Discover, and Visualize apps).
+
+### Example registration
+
+```ts
+import { ShareContext, ShareMenuItem } from 'src/plugins/share/public';
+
+plugins.share.register({
+  id: 'demo',
+  getShareMenuItems: (context) => [
+    {
+      panel: {
+        id: 'demo',
+        title: 'Panel title',
+        content: 'Panel content',
+      },
+      shareMenuItem: {
+        name: 'Demo list item (from share_example plugin)',
+      },
+    }
+  ],
+});
+```
+
+Now the "Demo list item" will appear under the "Share" menu in Discover and
+Dashboard applications.
+
+
+## Locators
+
+*Locators* are entities which given some *parameters* can navigate Kibana user
+to some deep link in Kibana or return back a full formatted URL.
+
+Kibana apps create locators and expose them from their plugin contracts. Then
+other plugins can use those locators to navigate deeply into Kibana applications.
+
+Locators work, both, on the server and browser. However, it is the responsibility
+of the app, which provides the locator, to register and expose the same locator
```

Comment:
worth putting this into a "warning" box to make it stand out, i think its very important that you register locators on both server and client and that one locator with its id is the same on both sides